Doppler.BillingUser/Controllers/BillingController.cs

```diff
@@ -251,6 +254,108 @@ public async Task<IActionResult> UpdateCurrentPaymentMethod(string accountname,
             }
         }
 
+        [Authorize(Policies.PROVISORY_USER)]
+        [HttpPut("/accounts/{accountname}/payments/reprocess")]
+        public async Task<IActionResult> Reprocess(string accountname)
+        {
+            ReprocessInvoiceResult result;
+
+            var user = await _userRepository.GetUserInformation(accountname);
+
+            var userBillingInfo = await _userRepository.GetUserBillingInformation(accountname);
+
+            var invoices = await _billingRepository.GetInvoices(user.IdUser);
+
+            if (invoices.Count == 0)
+            {
+                _logger.LogError("Invoices with accountname: {accountname} were not found.", accountname);
+                result = ReprocessInvoiceResult.Failed("Invoices not found");
+                return new NotFoundObjectResult(result);
+            }
+
+            if (userBillingInfo.PaymentMethod != PaymentMethodEnum.CC && userBillingInfo.PaymentMethod != PaymentMethodEnum.MP)
+            {
+                result = ReprocessInvoiceResult.Failed("Payment method has to be either Credit Card or Mercado Pago");
+
+                return new BadRequestObjectResult(result);
+            }
+
+            var invoicesResults = new List<ReprocessInvoicePaymentResultEnum>();
+
+            foreach (var invoice in invoices)
+            {
+                var reprocessResult = await ReprocessInvoicePayment(invoice, accountname, user, userBillingInfo);
+                invoicesResults.Add(reprocessResult);
+            }
+
+            // Checks whether all the invoices were process succesfully
+            if (invoicesResults.All(x => x.Equals(ReprocessInvoicePaymentResultEnum.Successful)))
+            {
+                _userRepository.UnblockAccountNotPayed(accountname);
+
+                result = ReprocessInvoiceResult.Success();
+
+                return new ObjectResult(result);
+            }
+
+            if (invoicesResults.Contains(ReprocessInvoicePaymentResultEnum.Successful))
+            {
+                _userRepository.UnblockAccountNotPayed(accountname);
+
+                result = ReprocessInvoiceResult.Failed("At least one of the invoices was succesfully reprocess");
+
+                return new ObjectResult(result);
+            }
+            else
+            {
+                result = ReprocessInvoiceResult.Failed("No invoice was reprocess succesfully");
+
+                return new ObjectResult(result);
+            }
```

Comment:
I would invert this:

```suggestion
            if (!invoicesResults.Contains(ReprocessInvoicePaymentResultEnum.Successful))
            {
                result = ReprocessInvoiceResult.Failed("No invoice was reprocess succesfully");

                return new ObjectResult(result);
            }

            _userRepository.UnblockAccountNotPayed(accountname);
                
            // Checks whether all the invoices were process succesfully
            if (invoicesResults.All(x => x.Equals(ReprocessInvoicePaymentResultEnum.Successful)))
            {
                result = ReprocessInvoiceResult.Success();
                return new ObjectResult(result);
            }
            else
            {
                result = ReprocessInvoiceResult.Failed("At least one of the invoices was succesfully reprocess");
                return new ObjectResult(result);
            }
```

Reply:
Done!